ui/src/components/auth/SignInForm.tsx

```diff
@@ -1,19 +1,21 @@
-import { Alert, Box, Button, Link, TextField } from "@mui/material";
-import NextLink from "next/link";
-import React, { useState } from "react";
+import { Alert, Box, Button, TextField } from "@mui/material";
+import { FormEvent, useState } from "react";
 
-interface SignInFormProps {
-  onSubmit: (email: string, password: string) => void;
+import Link from "@/components/Link";
+import { SignInWithPasswordMutationVariables } from "@/graphql/generated";
+
+type SignInFormProps = {
+  onSubmit: (args: SignInWithPasswordMutationVariables) => void;
```

Comment:
Ideally we should just use the generated types as the function signature, more inference less manual typing.

It's slightly annoying to convert object types into function args (tuple type) so I decided to just use the object type. It's actually nicer because this way you can just pass the generated mutation function directly in to `onSubmit` (see `sign-in.tsx`)

(There's also a `SignInWithPasswordInput` type but I think `SignInWithPasswordMutationVariables` is the better one because it's actually what's used in the mutation function, lmk if you know what the difference is—I think `Input` is used to compose mutations, and `MutationVariables` has `Exact<>` but seems like `Exact<>` [doesn't actually do anything](https://github.com/dotansimha/graphql-code-generator/issues/4577))

Reply:
Nice, fixed this in the other forms too.